### DISCUSSION TITLE
Can you create a similar utility that only manages Intune Mobile Apps and Devices, as well as Users and Groups?

### DISCUSSION THREAD

**Discussion by Brizzriel** [Micke-K/IntuneManagement#253] [Ideas]
I have not seen this request yet, so I thought I'd add it.

I look at your utility and it makes my head hurt. I believe it is the best of its kind I have ever seen, there are not many btw, that I've seen that is. 

I would like to have one that only manages mobile devices their policies, applications, profiles and configurations. It would also need to have access to at least Groups but the option for users would be nice too. 

I know that these things are already there its just that our security team wont allow "ALL" of it for some reason, but they will allow the pieces I need which are those I already listed.

**Comment by Micke-K** [comment]
Hello,

Thank you so much!

You can add the permissions you need and skip the ones you don't need. It will still work. You will get red/orange menu items for those that are missing permissions. And probably a few errors in the log but those can be ignored. 

Make sure the "Use Default Permissions" is checked in Settings. This will make sure it only uses the permissions assigned to the app and it won't prompt for all. 

Cheers!

**Reply by Brizzriel** [reply]
Thank you, I will try that and see what they say. I appreciate your hard work on this project. I do my best to write scripts and even bought the PowerShell Studio from Sapien Tech so I can make nice looking apps, but I am too ignorant at this time to figure it out well enough to get it to work the way I need it to. I you could tell me what sections I need for the Intune Mobile side of stuff only and the Users and Groups stuff. I know you probably don't have time for that, but I would appreciate it if you could at least point me in the right direction that would be awesome. Thank you again.

**Reply by Micke-K** [reply]
Hello,

The best way is probably to ask the security team to creat your own App registration in Entra and then add the permissions you need. You can then change the App id in Settings. 

Another way is to update the EndpointManager.psm1 script and remove all Add-ViewItem you don't need. There is one for each menu item. The problem with that is that you will have to do that again every time the file is updated. 

Cheers!

**Reply by Brizzriel** [reply]
That does sound irritating lol. Is there a way to add groups and set it to where you have to enter a device name or serial number and then be able to select an app or cnfg profile they want on the device and then get a list of the groups involved with that item. They do not have access to the group itself but can add the entered device to or remove it from a listed group. The reason for this setup is that we do not want to give our on-site technicians access to groups just the ability to add or remove a device from one via the GUI.

Let me know if you have any ideas on how to set this up. It would be awesome if you could add it as option in your app  :-D

**Reply by Micke-K** [reply]
Hello,

Not sure I follow. The tool can't really do this since the permissions will be for an API and not a specific object (eg a Group) for that API. There are workaround eg use something like App functions, Logic Apps for adding members. This would be a complex solution too much to implement in the tool. The part of listing policies and get assigned groups is possible. The "Intune Assignments" tool can do parts of this. This only supports exported files at the moment.

Cheers!
